impl/src/main/java/com/sun/faces/context/flash/ELFlash.java

```diff
@@ -1289,7 +1289,7 @@ void expireNext_MovePreviousToNext() {
                 flashMap.clear();
                 // remove it from the flash
                 innerMap.remove(nextRequestFlashInfo.getSequenceNumber() + "");
-                nextRequestFlashInfo = null;
+                expireNext();
```

Comment:
Is this something that was previously in our fork that we're cherry-picking here? If so, we should keep that as a separate commit so it stands out as an additional commit in our branch.

Reply:
That is a cherry-pick, yes.